messaging/broker.go

```diff
@@ -1017,6 +1009,19 @@ func NewTopicReader(path string, index uint64, streaming bool) *TopicReader {
 	}
 }
 
+// Seek seeks to a position the current segment.
+func (r *TopicReader) Seek(offset int64, whence int) (int64, error) {
+	assert(whence == os.SEEK_CUR, "topic reader can only seek to a relative position")
```

Comment:
If someone passes a non-`SEEK_CUR` in the future I want to catch it immediately. If I silently override it then it'll be a pain to debug.